logic/src/commonTest/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepositoryTest.kt

```diff
@@ -161,14 +162,115 @@ class ConnectionRepositoryTest {
             .wasInvoked(once)
     }
 
+    @Test
+    fun givenAConnectionRequestUpdate_WhenSendingAConnectionStatusValid_thenTheConnectionShouldBePersisted() = runTest {
+        // given
+        val userId = NetworkUserId("user_id", "domain_id")
+        given(connectionApi)
+            .suspendFunction(connectionApi::updateConnection)
+            .whenInvokedWith(eq(userId), eq(ConnectionStateDTO.ACCEPTED))
+            .then { _, _ -> NetworkResponse.Success(connection1, mapOf(), 200) }
+        given(conversationDAO)
+            .suspendFunction(conversationDAO::updateOrInsertOneOnOneMemberWithConnectionStatus)
+            .whenInvokedWith(eq(UserIDEntity(userId.value, userId.domain)), any(), any())
+            .then { _, _, _ -> return@then }
+
+        val result = connectionRepository.updateConnectionStatus(UserId(userId.value, userId.domain), ConnectionState.ACCEPTED)
+        result.shouldSucceed()
+
+        // then
+        verify(connectionApi)
+            .suspendFunction(connectionApi::updateConnection)
+            .with(eq(userId), eq(ConnectionStateDTO.ACCEPTED))
+            .wasInvoked(once)
+        verify(conversationDAO)
+            .suspendFunction(conversationDAO::updateOrInsertOneOnOneMemberWithConnectionStatus)
+            .with(any(), any(), any())
+            .wasInvoked(once)
+    }
+
+    @Test
+    fun givenAConnectionRequestUpdate_WhenSendingAConnectionStatusInvalid_thenTheConnectionShouldThrowAnError() = runTest {
+        // given
+        val userId = NetworkUserId("user_id", "domain_id")
+        given(connectionApi)
+            .suspendFunction(connectionApi::updateConnection)
+            .whenInvokedWith(eq(userId), any())
+            .then { _, _ -> NetworkResponse.Success(connection1, mapOf(), 200) }
```

Comment:
Could this be a typo from copy/pasting, as this return a `400` instead of a `200`? If so, I don't really understand how this test is passing below when running `result.shouldFail()`

Reply:
It's passing because, this is not an error related to network API perse, but more an error regarding the connection status (trying to update via user interaction to states other than ACCEPTED, IGNORED, CANCELLED) it will be triggered on the repository layer